datafusion/core/src/physical_optimizer/pruning.rs

```diff
@@ -344,11 +451,10 @@ impl RequiredStatColumns {
 
         // only add statistics column if not previously added
         if need_to_insert {
-            let stat_field = Field::new(
-                stat_column.name(),
-                field.data_type().clone(),
-                field.is_nullable(),
-            );
+            // may be null if statistics are not present
```

Comment:
A non-nullable column may appear as NULL in the statistics if the min or max values are not known